util/topgen/merge.py

```diff
@@ -637,38 +656,12 @@ def amend_resets(top):
     for intf in top['exported_rsts']:
         top['inter_module']['external']['{}.resets_{}'.format(
             rstmgr_name, intf)] = "rsts_{}".format(intf)
-    """Discover the full path and selection to each reset connection.
-       This is done by modifying the reset connection of each end point.
-    """
-    for end_point in top['module'] + top['memory'] + top['xbar']:
-        for port, net in end_point['reset_connections'].items():
-            reset_path = lib.get_reset_path(net, end_point['domain'],
-                                            top['resets'])
-            end_point['reset_connections'][port] = reset_path
-
-    # reset paths are still needed temporarily until host only modules are properly automated
-    reset_paths = OrderedDict()
-    reset_hiers = top["resets"]['hier_paths']
-
-    for reset in top["resets"]["nodes"]:
-        if "type" not in reset:
-            log.error("{} missing type field".format(reset["name"]))
-            return
-
-        if reset["type"] == "top":
-            reset_paths[reset["name"]] = "{}rst_{}_n".format(
-                reset_hiers["top"], reset["name"])
-
-        elif reset["type"] == "ext":
-            reset_paths[reset["name"]] = reset_hiers["ext"] + reset['name']
-        elif reset["type"] == "int":
-            log.info("{} used as internal reset".format(reset["name"]))
-        else:
-            log.error("{} type is invalid".format(reset["type"]))
 
-    top["reset_paths"] = reset_paths
+    # reset class objects
+    top["resets"] = top_resets
 
-    return
+    # The original resets dict is transformed to the reset class
+    assert isinstance(top["resets"], Resets)
```

Comment:
@rswarbrick ah i moved the isinstance check here. 
is this what you meant?

Reply:
Oh! Sorry, I meant that it makes sense to have the check in the topgen.py code because there we're grabbing some random dictionary entry and using it as a `Resets` object. As well as a bit of error checking, that serves as a sort of documentation to the next person that reads it about what the object is supposed to be. But no big deal: if we end up doing more "classification" in the topgen code, the code should get clearer over time anyway.

Reply:
oooo i see.
that makes sense. I'll add that in.